src/main/java/org/openmrs/addonindex/scheduled/FetchDetailsToIndex.java

```diff
@@ -111,7 +111,9 @@ void getDetailsAndIndex(AddOnToIndex toIndex) throws Exception {
 			if (fetchExtraDetails) {
 				fetchExtraDetailsForEachVersion(toIndex, infoAndVersions);
 			}
-			
+
+			infoAndVersions.setDetailsBasedOnLatestVersion(infoAndVersions.getVersion(
```

Comment:
Move the logic for choosing the latest version into AddOnInfoAndVersion. Actually, it seems worth adding getLatestAddOnVersion() would make sense, since I feel like we've tried to do this a couple of times now.